File: 3225-length-of-longest-subarray-with-at-most-k-frequency/length-of-longest-subarray-with-at-most-k-frequency.cpp
class Solution {
public:
    int maxSubarrayLength(vector<int>& v, int k) {
        unordered_map<int,int> m ;
        int n = v.size();
        int i =0,j=0;
        int ans =0;
        while(i<=j&& j<n){
            m[v[j]]++;
            if(m[v[j]]>k){
                while(i<=j && m[v[j]]>k){
                    if(m[v[i]]==1){
                        m.erase(v[i]);
                    }
                    else{
                        m[v[i]]--;
                    }
                    i++;
                }
            }
            ans = max(ans,j-i+1);
            j++;
        }
        return ans ;
    }
};

File: 3225-length-of-longest-subarray-with-at-most-k-frequency/length-of-longest-subarray-with-at-most-k-frequency.cs
public class Solution {
    public int MaxSubarrayLength(int[] v, int k)
{
    Dictionary<int, int> m = new Dictionary<int, int>();

    int n = v.Length;
    int i = 0, j = 0;
    int ans = 0;

    while (i <= j && j < n)
    {
        if (!m.ContainsKey(v[j]))
            m[v[j]] = 0;

        m[v[j]]++;

        if (m[v[j]] > k)
        {
            while (i <= j && m[v[j]] > k)
            {
                m[v[i]]--;

                if (m[v[i]] == 0)
                    m.Remove(v[i]);

                i++;
            }
        }

        ans = Math.Max(ans, j - i + 1);
        j++;
    }

    return ans;
}
}